### PR TITLE
Add some ICEs

### PR DESCRIPTION
**Pull request by fanninpm**
- Add 86528
- Add 86535
- Add 86642
- Add 86703
- Add 86719
- Add 86720
- Add 86756
- Add 86784
- Add 86800
- Add 86820
- Add 86850

### CODE REVIEW

File: ices/86535.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+struct Bar<const N: &'static ()>;
+impl<const N: &'static ()> Foo for Bar<N> {
+    const ASSOC_C: usize = 3;
+
+    fn foo() where [u8; Self::ASSOC_C]: {
+        let _: [u8; Self::ASSOC_C] = loop {};
+    }
+}
```
Comment (JohnTitor) [inline ices/86535.rs:17]:
```suggestion
#![feature(box_syntax)]
fn main() {
    let _: Box<[isize]> = box { loop {} };
}
```

File: ices/86642.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#![feature(impl_trait_in_bindings)]
```
Comment (JohnTitor) [inline ices/86642.rs:1]: It seems like that this is likely a duplicate of the mentioned issue but let's see it with glacier.

File: ices/86800.rs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+impl Context {
+    async fn do_transaction<O>(
+        &self, f: impl FnOnce(&mut dyn Transaction) -> TransactionFuture<O>
+    ) -> TransactionResult<O>
+    {
+        let mut conn = Connection {};
+        let mut transaction = TestTransaction { conn: &mut conn };
+        f(&mut transaction).await
+    }
+}
```
Comment (JohnTitor) [inline ices/86800.rs:40]:
```suggestion
#![feature(min_type_alias_impl_trait)]
use std::future::Future;

struct Connection {}

trait Transaction {}

struct TestTransaction<'conn> {
    conn: &'conn Connection,
}

impl<'conn> Transaction for TestTransaction<'conn> {}

struct Context {}

type TransactionResult<O> = Result<O, ()>;

type TransactionFuture<'__, O> = impl '__ + Future<Output = TransactionResult<O>>;

fn execute_transaction_fut<'f, F, O>(
    f: F,
) -> impl FnOnce(&mut dyn Transaction) -> TransactionFuture<O>
where
    F: FnOnce(&mut dyn Transaction) -> TransactionFuture<O> + 'f,
{
    f
}

impl Context {
    async fn do_transaction<O>(
        &self,
        f: impl FnOnce(&mut dyn Transaction) -> TransactionFuture<O>,
    ) -> TransactionResult<O> {
        let mut conn = Connection {};
        let mut transaction = TestTransaction { conn: &mut conn };
        f(&mut transaction).await
    }
}

```

File: ices/86820.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
```
Comment (JohnTitor) [inline ices/86820.sh:1]: This doesn't have to be run as an actual crate. All it needs is to place `assert!` on `fn main()`.

**Review by JohnTitor — CHANGES_REQUESTED** [review]
Some examples need to be tweaked manually. Let me do and push it.

**Comment by JohnTitor** [comment]
I'm a bit concerned that one example is very large but it doesn't affect the check time much, so I'm going to accept it.

**Comment by hellow554** [comment]
@JohnTitor you mean https://github.com/rust-lang/rust/issues/86784#issuecomment-872886518 ?

Yeah, it's crazy how big it is, but surprisingly fast to compile, mostly because it's only assignments. Thanks for adding it ;)

**Pull request merged**